server/journeys/referral/involvement/schemas.ts

```diff
@@ -0,0 +1,50 @@
+import z from 'zod'
+import { Request } from 'express'
+import type CsipApiService from '../../../services/csipApi/csipApiService'
+
+const csrfSchema = z.object({
+  _csrf: z.string().optional(),
+})
+const involvementTypeErrorMsg = `Select how the prisoner has been involved in the behaviour`
+const isStaffAssaultedErrorMsg = `Select if any staff were assaulted as a result of the behaviour or not`
+const emptyStaffMemberErrorMsg = `Enter the staff member's name`
+const tooLongStaffMemberErrorMsg = `Staff member's name must be 240 characters or less`
+
+export const schemaFactory = (csipApiService: CsipApiService) => async (req: Request) => {
+  const incidentInvolvementMap = new Map(
+    (await csipApiService.getReferenceData(req, 'incident-involvement')).map(itm => [itm.code, itm]),
+  )
+
+  return csrfSchema
+    .merge(
+      z.object({
+        involvementType: z.string({ message: involvementTypeErrorMsg }).transform((val, ctx) => {
+          if (!incidentInvolvementMap.has(val)) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
+              message: involvementTypeErrorMsg,
+            })
+            return z.NEVER
+          }
+          return incidentInvolvementMap.get(val)!
+        }),
+        staffAssaulted: z
+          .string({ message: isStaffAssaultedErrorMsg })
+          .trim()
+          .refine(val => ['false', 'true'].includes(val), { message: isStaffAssaultedErrorMsg }),
+        assaultedStaffName: z.string().max(240, { message: tooLongStaffMemberErrorMsg }),
```

Comment:
Do you think it's worth having a boundary test for this max string length? I would normally have tests for all possible validation errors

Reply:
Oops, yes it is, I meant to do that. Adding now!

Reply:
Fixed.